addons/binding/org.openhab.binding.keba/src/main/java/org/openhab/binding/keba/handler/KeContactBroadcastListener.java

```diff
@@ -0,0 +1,196 @@
+package org.openhab.binding.keba.handler;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.nio.ByteBuffer;
+import java.nio.channels.ClosedChannelException;
+import java.nio.channels.DatagramChannel;
+import java.nio.channels.SelectionKey;
+import java.nio.channels.Selector;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.core.common.ThreadPoolManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class KeContactBroadcastListener {
+
+    private static final String THING_HANDLER_THREADPOOL_NAME = "thingHandler";
+    public static final int LISTENER_PORT_NUMBER = 7090;
+    public static final int LISTENING_INTERVAL = 100;
+
+    private DatagramChannel broadcastChannel = null;
+    private SelectionKey broadcastKey = null;
+    private Selector broadcastListener;
+    private ScheduledFuture<?> broadcastJob;
+    private boolean isStarted = false;
+    protected final ScheduledExecutorService scheduler = ThreadPoolManager
+            .getScheduledPool(THING_HANDLER_THREADPOOL_NAME);
+    private Set<KeContactHandler> listeners = Collections.synchronizedSet(new HashSet<KeContactHandler>());
+
+    final Logger logger = LoggerFactory.getLogger(KeContactBroadcastListener.class);
+
+    public void start() {
+
+        if (!isStarted) {
+            logger.info("Starting the Keba broadcast listener");
+            try {
+                broadcastChannel = DatagramChannel.open();
+                broadcastChannel.socket().bind(new InetSocketAddress(LISTENER_PORT_NUMBER));
+                broadcastChannel.configureBlocking(false);
+
+                try {
+                    broadcastListener = Selector.open();
+                } catch (IOException e) {
+                    logger.error("An exception occurred while registering the selector: '{}'", e.getMessage());
+                }
+
+                logger.info("Listening for incoming data on {}", broadcastChannel.getLocalAddress());
+
+                synchronized (broadcastListener) {
+                    broadcastListener.wakeup();
+                    try {
+                        broadcastKey = broadcastChannel.register(broadcastListener, broadcastChannel.validOps());
+                    } catch (ClosedChannelException e1) {
+                        logger.error("An exception occurred while registering a selector: {}", e1.getMessage());
+                    }
+                }
+
+                if (broadcastJob == null || broadcastJob.isCancelled()) {
+                    broadcastJob = scheduler.scheduleWithFixedDelay(broadcastRunnable, 0, LISTENING_INTERVAL,
```

Comment:
I think it will be necessary in situation where there are bunch of charging stations. the Runnable first queries a selector, so, in that sense it is light enough, I think